testslide/lib.py

```diff
@@ -82,17 +82,33 @@ def _validate_function_signature(
                 expected_type = argspec.annotations.get(argname)
                 if not expected_type:
                     continue
+
+                if '~' in str(expected_type):
```

Comment:
Is there a more thorough way of detecting this? Eg: `isinstance`?

Reply:
The problem is that it can be nested, like `Type[~Something]`, `Optional[Tuple[int, Tuple[str, ~Something]]]`, and I did not want to have to implement the unwrapping of that, but as `str` already does and `~` is kinda unique, seems like a good trade-off.

But yes, I think that eventually we will have to unwrap that, at that point we can use that here instead.

Reply:
That's reasonable for the time being, NP.